gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go

```diff
@@ -905,12 +905,10 @@ func main() {
 	provName := provisionerName
 	provEnvName := os.Getenv(provisionerNameKey)
 
-	// Precedence is given for ProvisionerNameKey
-	if provEnvName != "" && *id != "" {
+	// Precedence is given for ProvisionerNameKey if both are set
+	if provEnvName != "" && *id != "" || provEnvName != "" {
```

Comment:
The logic currently reads like this:

```
if (a && b) || a {
  use ENV
} else if b {
  use cli
}
```

If we want to keep logic as in the table 

| ENV set | CLI set | Name of the provisioner|
| ------------- | ------------- | ------------- |
| No | No | Default |
| Yes | No | ENV |
| No | Yes | CLI |
| Yes | Yes | ENV |

Then I suggest we simplify the code as
```
if a {
  use ENV
} else if b {
  use cli
}
```